include/linux/fs.h

```diff
@@ -102,6 +102,22 @@ typedef int (dio_iodone_t)(struct kiocb *iocb, loff_t offset,
 /* called from RCU mode, don't block */
 #define MAY_NOT_BLOCK		0x00000080
 
+#if CONFIG_TRUENAS
+/*
+ * Extended NFSv41 write permissions. These are used selectively
+ * for permissions checks where NFSv4 ACL handling is
+ * more nuanced than a simple POSIX permissions.
+ */
+#define MAY_WRITE_NAMED_ATTRS	0x00000100
```

Comment:
[nit]: please feel free to ignore any of my nits, but I find that having the hex values aligned at the next logical tab makes the code look cleaner.  Just my $0.02, fwiw

Reply:
I'll do that.